core/app/models/spree/credit_card.rb

```diff
@@ -46,12 +46,15 @@ def self.default
 
     def default
       Spree::Deprecation.warn("CreditCard.default is deprecated. Please use user.wallet.default_wallet_payment_source instead.", caller)
-      user.wallet.default_wallet_payment_source.payment_source == self
+      return false if user.nil?
+      user.wallet.default_wallet_payment_source.try!(:payment_source) == self
     end
 
     def default=(set_as_default)
       Spree::Deprecation.warn("CreditCard.default= is deprecated. Please use user.wallet.default_wallet_payment_source= instead.", caller)
-      if set_as_default # setting this card as default
+      if user.nil?
```

Comment:
Does this work reasonably with the current default behavior and guest checkouts?

https://github.com/solidusio/solidus/blob/master/core/app/models/spree/wallet/default_payment_builder.rb#L14

Its tough to trace all the way through easily (with all the aliases and whatnot) but I think we'll get the default set to nil if there is no user? Though none of our specs seem to be failing...